src/types.ts

```diff
@@ -77,7 +77,10 @@ export type AppState = {
   // (e.g. text element when typing into the input)
   editingElement: NonDeletedExcalidrawElement | null;
   editingLinearElement: LinearElementEditor | null;
-  activeTool: { type: typeof SHAPES[number]["value"] | "eraser" };
+  activeTool: {
+    type: typeof SHAPES[number]["value"] | "eraser";
+    typeBeforeEraser?: typeof SHAPES[number]["value"] | "eraser";
```

Comment:
`typeBeforeEraser` could be confusing. Let's call it `lastActiveToolBeforeEraser`. It doesn't matter the "type" doesn't quite correlate with the naming, as long as the naming is more clear.

Also, the `| "eraser"` should be removed from here.

Reply:
I'll rename the variable.

I also wanted to remove ` | "eraser"`, but then I got a type error here:

https://github.com/excalidraw/excalidraw/blob/db6b335373a9c543304925331f4565acedbaaf90/src/actions/actionCanvas.tsx#L308-L310

Reply:
Strangely, when `lastActiveToolBeforeEraser` does not include "eraser" in the type definition I get a type error on the above code when using `isEraserActive(appState)`, but not when I use this expression directly `appState.activeTool.type === "eraser"`